src/backend/base/langflow/base/data/kb_utils.py

```diff
@@ -102,3 +108,47 @@ def compute_bm25(documents: list[str], query_terms: list[str], k1: float = 1.2,
         scores.append(doc_score)
 
     return scores
+
+
+async def _get_current_user(langflow_api_key: str | None = None) -> User:
+    """Get the current user based on the provided API key or create a new user."""
+    async for db in get_session():
+        if langflow_api_key:
+            current_user = await get_current_user(
+                token="",
+                query_param=langflow_api_key,
+                header_param="",
+                db=db,
+            )
+        else:
+            user_id, _ = await create_user_longterm_token(db)
+            current_user = await get_user_by_id(db, user_id)
```

Comment:
I'm confused by this. One important aspect is that Components have `self.user_id` which can help you get the user.

Reply:
Good call, this was actually what I was looking for - trying to get the current user - and the approach i took was a bad one (check for an API key, if it doesnt exist, use the super user).

I'll fix this. appreciate it @ogabrielluiz